### PR TITLE
Feature/scalarmult

### PR DESCRIPTION
**Pull request by IamFlowZ**

### CODE REVIEW

**Comment by IamFlowZ** [comment]
#38

File: android/src/main/java/org/libsodium/rn/RCTSodiumModule.java [hunk 1]
```diff
@@ -476,11 +476,32 @@ public void crypto_scalarmult_base(final String n, final Promise p) {
           p.resolve(Base64.encodeToString(q,Base64.NO_WRAP));
       }
     }
-    catch (Throwable t) {
+    catch (Throwable t) {
```
Comment (lyubo) [inline android/src/main/java/org/libsodium/rn/RCTSodiumModule.java]: Please, revert this change as it is not relevant to new functionality

File: android/src/main/java/org/libsodium/rn/RCTSodiumModule.java [hunk 2]
```diff
@@ -476,11 +476,32 @@ public void crypto_scalarmult_base(final String n, final Promise p) {
...
       }
     }
-    catch (Throwable t) {
+    catch (Throwable t) { 
       p.reject(ESODIUM,ERR_FAILURE,t);
     }
   }
 
+  @ReactMethod
+  public void crypto_scalarmult(final String n, final String q, final Promise p) {
```
Comment (lyubo) [inline android/src/main/java/org/libsodium/rn/RCTSodiumModule.java]: In native crypto_scalarmult n and p are the input parameters, while q is the output. Please correct parameter names of this method to reflect their purpose as in the original.

File: index.d.ts [hunk 3]
```diff
@@ -203,6 +203,10 @@ declare module "react-native-sodium" {
    * The key pair can be deterministically derived from a single key seed.
    */
   export function crypto_scalarmult_base(secretKey: string): Promise<string>;
+  export function crypto_scalarmult(
+    publicKey: string,
```
Comment (lyubo) [inline index.d.ts]: In relation to previous comment, I think the first argument should be 'n', which contains the private key (as in the libsodium documentation, see https://doc.libsodium.org/advanced/scalar_multiplication), and the second one is 'q'.

File: ios/RCTSodium/RCTSodium.m [hunk 4]
```diff
@@ -373,6 +373,19 @@ + (BOOL)requiresMainQueueSetup
     resolve([[NSData dataWithBytesNoCopy:q length:sizeof(q) freeWhenDone:NO] base64EncodedStringWithOptions:0]);
 }
 
+RCT_EXPORT_METHOD(crypto_scalarmult:(NSString*)n p:(NSString*)p resolve:(RCTPromiseResolveBlock)resolve reject:(RCTPromiseRejectBlock)reject)
```
Comment (lyubo) [inline ios/RCTSodium/RCTSodium.m]: Parameter names as in my comments on java interface.

**Review by lyubo — CHANGES_REQUESTED** [review]
Please see my comments on files.

**Pull request merged**